### PR TITLE
Add hexagonal containers

### PR DESCRIPTION
**Pull request by harshavb**
@bdach was unable to make changes in #4016, possibly due to PRing from master. This pull request attempts to fix this issue. Once again, all credit goes to Poyo for writing the code.

### CODE REVIEW

**Comment by frenzibyte** [comment]
The title shouldn't have anything irrelevant in it, rename back to `Add hexagonal containers` or otherwise

File: osu.Framework/Graphics/Containers/HexagonalContainer.cs [hunk 1]
```diff
@@ -0,0 +1,144 @@
...
+
+namespace osu.Framework.Graphics.Containers
+{
+    public class HexagonalContainer : HexagonalContainer<Drawable> { }
+
+    public class HexagonalContainer<T> : BufferedContainer<T>
+        where T : Drawable
+    {
+        private static readonly float sin_pi_over_3 = (float)Math.Sin(Math.PI / 3);
+        private static readonly float tan_pi_over_3 = (float)Math.Tan(Math.PI / 3);
```
Comment (turbedi) [inline osu.Framework/Graphics/Containers/HexagonalContainer.cs:24]:
```suggestion
        private static readonly float sin_pi_over_3 = MathF.Sin(MathF.PI / 3);
        private static readonly float tan_pi_over_3 = MathF.Tan(MathF.PI / 3);
```

File: osu.Framework/Graphics/Primitives/Hexagon.cs [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+using System.Runtime.InteropServices;
+
+namespace osu.Framework.Graphics.Primitives
+{
+    [StructLayout(LayoutKind.Sequential)]
+    public class Hexagon : IConvexPolygon, IEquatable<Hexagon>
+    {
+        // Note: Do not change the order of vertices. They are ordered in screen-space counter-clockwise fashion.
+        // See: IPolygon.GetVertices()
+        private static readonly float SQRT_3 = (float)Math.Sqrt(3);
```
Comment (turbedi) [inline osu.Framework/Graphics/Primitives/Hexagon.cs:16]:
```suggestion
        private static readonly float SQRT_3 = MathF.Sqrt(3);
```

**Comment by harshavb** [comment]
Closing due to lack of interest + broken hexagon rotations

**Pull request closed without merging**